app/src/shared/hooks/useAccountAddress.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+
+import { useState, useEffect } from 'react'
+import { getWeb3 } from '$shared/web3/web3Provider'
+import Web3Poller from '$shared/web3/web3Poller'
+import useIsMounted from '$shared/hooks/useIsMounted'
+
```

Comment:
@mondoreale I'm getting around to the idea that a lightweight version like this is needed. We should also add the error listener so that account is reset if user closes the metamask window.

```
useEffect(() => {
    (async () => {
            try {
                return await getWeb3().getDefaultAccount()
            } catch (e) {
                return undefined
            }
        })().then((addr) => {
            if (isMounted()) {
                setAddress(addr)
            }
        })

        const setLocked = () => {
            if (!isMounted()) { return }
            setAddress(null)
        }

        Web3Poller.subscribe(Web3Poller.events.ACCOUNT, setAddress)
        Web3Poller.subscribe(Web3Poller.events.ACCOUNT_ERROR, setLocked)

        return () => {
            Web3Poller.unsubscribe(Web3Poller.events.ACCOUNT, setAddress)
            Web3Poller.unsubscribe(Web3Poller.events.ACCOUNT_ERROR, setLocked)
        }
}, [isMounted])
```